price-feeder/oracle/provider/osmosis.go

```diff
@@ -170,15 +170,20 @@ func (p OsmosisProvider) GetCandlePrices(pairs ...types.CurrencyPair) (map[strin
 			return nil, fmt.Errorf("failed to unmarshal Osmosis response body: %w", err)
 		}
 
+		staleTime := PastUnixTime(providerCandlePeriod)
+
 		candlePrices := []types.CandlePrice{}
 		for _, responseCandle := range candlesResp {
-			closeStr := fmt.Sprintf("%f", responseCandle.Close)
-			volumeStr := fmt.Sprintf("%f", responseCandle.Volume)
-			candlePrices = append(candlePrices, types.CandlePrice{
-				Price:     sdk.MustNewDecFromStr(closeStr),
-				Volume:    sdk.MustNewDecFromStr(volumeStr),
-				TimeStamp: responseCandle.Time,
-			})
+			if staleTime < responseCandle.Time {
+				closeStr := fmt.Sprintf("%f", responseCandle.Close)
+				volumeStr := fmt.Sprintf("%f", responseCandle.Volume)
+				candlePrices = append(candlePrices, types.CandlePrice{
+					Price:  sdk.MustNewDecFromStr(closeStr),
+					Volume: sdk.MustNewDecFromStr(volumeStr),
+					// convert osmosis timestamp seconds -> milliseconds
+					TimeStamp: SecondsToMilli(responseCandle.Time),
+				})
+			}
```

Comment:
This is not needed, just a suggestion

```suggestion
			if staleTime >= responseCandle.Time {
			        continue
			}
			closeStr := fmt.Sprintf("%f", responseCandle.Close)
			volumeStr := fmt.Sprintf("%f", responseCandle.Volume)
			candlePrices = append(candlePrices, types.CandlePrice{
			Price:  sdk.MustNewDecFromStr(closeStr),
			Volume: sdk.MustNewDecFromStr(volumeStr),
			// convert osmosis timestamp seconds -> milliseconds
			TimeStamp: SecondsToMilli(responseCandle.Time),
			})
```